syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.leveldb";
option java_outer_classname = "LevelDBProto";
option objc_class_prefix = "HLW";

package database;

service Database {
  rpc Get (GetRequest) returns (GetResponse) {}
  rpc Put (PutRequest) returns (PutResponse) {}
  rpc PutResult (PutResultRequest) returns (PutResultResponse) {}
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 2;
}

message PutRequest {
  int64 seqnum = 1;
  string key = 2;
  string value = 3;
}

message PutResponse {

}

// I should perhaps use a different proto file for this, but meh
message PutResultRequest {
  int64 seqnum = 1;
  bool success = 2;
}

message PutResultResponse {

}
